### DISCUSSION TITLE
gorm.ErrDuplicatedKey not working as intended

### DISCUSSION THREAD

**Discussion by thilak009** [go-gorm/gorm#6447] [Q&A]
I have upgraded gorm to version `1.25.2` and added the necessary config as mentioned in the [documentation](https://gorm.io/docs/error_handling.html#Dialect-Translated-Errors) for using dialect translated errors, but it does not seem to work, 
DB connection
```
db, err = gorm.Open(postgres.Open(DSN), &gorm.Config{TranslateError: true})
```
the check
```
if errors.Is(err, gorm.ErrDuplicatedKey) {
    return errors.New("duplicate key error")
}
```
the condition results in false
the error that i see in logs
```
ERROR: duplicate key value violates unique constraint "accounts_pkey" (SQLSTATE 23505)
```
am i missing something? any help would be appreciated 🙏🏻

**Comment by EByrdS** [comment]
It seems to work for me.

**Comment by ZiRunHua** [comment]
That happened to me too so I'm backtracking 1.24.6 for now

**Comment by thilak009** [comment]
i ended up checking for the exact error code instead, i believe the latest version solves the issue, but i haven't been able to find realease notes to check for breaking changes for upgrade
```
var p *pgconn.PgError
if errorsInBuilt.As(err, &p) && p.Code == "23505" {
    return errors.BadRequest("account already exists")
}
```

**Reply by ZiRunHua** [reply]
I checked the git history of gorm and found that the error section of gorm was added on March 24 with a configuration item TranslateError which defaults to false, so now I need to set TranslateError to true when initializing gorm.
`gorm.Config{
    TranslateError: true,
}`

**Reply by dangquyitt** [reply]
It don't work for me. Currently I'm using https://github.com/go-gorm/postgres

**Reply by baohavan** [reply]
gorm.Config{ TranslateError: true, }

It worked for me

**Reply by climberlwz** [reply]
It worked for me

**Reply by AuroraTea** [reply]
`gorm.Config{ TranslateError: true, }` won't output which column was duplicated 😫

**Comment by jasonlimantoro** [comment]
Upgrading to `1.25.12` resolved the issue.

I had the same issue in `1.25.2` using `mysql`. It made no difference whether the `TranslateError` was true or false (unset).
